build/tools/ExtractRuleDescFromJson/RuleDescExtractor.cs

```diff
@@ -126,24 +167,32 @@ private static string EnsureHtmlIsXml(string html)
         return xml;
     }
 
-    private void SaveRuleFile(Rule rule, string xml)
+    private void SaveRuleFile(RuleInfo slvsRuleInfo)
     {
-        var fullPath = CalculateRuleFileName(rule);
+        var fullPath = CalculateRuleFileName(slvsRuleInfo);
         Logger.LogPartialMessage($" {Path.GetFileName(fullPath)}");
-        File.WriteAllText(fullPath, xml);
+
+        var options = new JsonSerializerOptions
+        {
+            PropertyNameCaseInsensitive = true,
+            WriteIndented = true
+        };
+
+        var data = JsonSerializer.Serialize(slvsRuleInfo, options);
+        File.WriteAllText(fullPath, data);
     }
 
-    private string CalculateRuleFileName(Rule rule)
+    private string CalculateRuleFileName(RuleInfo slvsRule)
     {
         // e.g. "S123.desc"
-        var colonPos = rule.Key.IndexOf(':');
+        var colonPos = slvsRule.RuleKey.IndexOf(':');
         if (colonPos == -1)
         {
-            throw new InvalidOperationException("Invalid rule key: " + rule.Key);
+            throw new InvalidOperationException("Invalid rule key: " + slvsRule.RuleKey);
         }
 
-        var ruleKeyWithoutLanguage = rule.Key.Substring(colonPos + 1);
```

Comment:
I changed the variable name `ruleKeyWithoutLanguage` since it was misleading - it's the repo key, not the language key.